tests/testthat/test-expect-shape.R

```diff
@@ -0,0 +1,51 @@
+test_that("shape computed correctly", {
+  # equivalent to expect_length
+  expect_success(expect_shape(1, 1))
+  expect_failure(expect_shape(1, 2), "has length 1, not length 2.")
+  expect_success(expect_shape(1:10, 10))
+  expect_success(expect_shape(letters[1:5], 5))
+
+  # testing dim()
+  expect_success(expect_shape(matrix(nrow = 5, ncol = 4), c(5L, 4L)))
+  expect_failure(expect_shape(matrix(nrow = 6, ncol = 3), c(6L, 2L)))
```

Comment:
A few of these should use `expect_snapshot_failure()` so we can see the failure messages.

Reply:
I don't have a good sense of when to use one vs. the other; at least one `snapshot_failure()` for each of the arguments makes sense to me superficially. I can add more plain `expect_failure()` if you see fit.